File: typecast-ruby/examples/tts.rb
require "typecast"

client = Typecast::Client.new
response = client.text_to_speech(
  Typecast::Models::TTSRequest.new(
    voice_id: "tc_60e5426de8b95f1d3000d7b5",
    text: "Hello from Typecast Ruby.",
    model: Typecast::Models::TTS_MODEL_V30,
    language: "eng",
    output: Typecast::Models::Output.new(audio_format: "wav")
  )
)

File.binwrite("hello.wav", response.audio_data)

File: typecast-dart/example/tts.dart
import 'dart:io';

import 'package:typecast_dart/typecast_dart.dart';

Future<void> main() async {
  final client = TypecastClient();
  final response = await client.textToSpeech(
    const TtsRequest(
      voiceId: 'tc_60e5426de8b95f1d3000d7b5',
      text: 'Hello from Typecast Dart.',
      model: TtsModel.ssfmV30,
      language: LanguageCode.eng,
      output: Output(audioFormat: AudioFormat.wav),
    ),
  );
  await File('hello.wav').writeAsBytes(response.audioData);
}
